m2cgen/interpreters/utils.py

```diff
@@ -7,3 +10,13 @@
 def get_file_content(path):
     with open(path) as f:
         return f.read()
+
+
+@lru_cache(maxsize=16)
```

Comment:
Does it make sense to put a larger number here to accommodate for future AST extension? I'd even suggest to make it unbounded if possible, since the total number of expressions in AST will always be pretty limited. Otherwise perhaps we can set it to something like 32.

Reply:
I think that for the sake of efficiency it should be as small as possible. I even thought to set `maxsize=8`.
For future AST extension we can easily update the `maxsize` here while adding new expressions.

Reply:
Yeah, my concern is that one should always keep this number on the back of their minds when adding new expression. Why  do you say it should be as small as possible though? The cache is super cheap in this case, isn't it? It's if we just had a map with 16 pointer to string pairs.

Reply:
> Yeah, my concern is that one should always keep this number on the back of their minds when adding new expression. 

Agree! It is not so convenient. Driving in a such direction we can simply replace two these functions with manual `dict`.

But according to the source code, `lru_cache` is based on linked list, not hashmap
https://github.com/python/cpython/blob/81a5fc38e81b424869f4710f48e9371dfa2d3b77/Lib/functools.py#L781

So I believe size matters here.

Reply:
OK, maybe I'm overoptimizing things. I've proposed new changes to this PR so that we won't have to care about keeping `maxsize` actual. Now it has enough capacity automatically. Please let me know what you think about it.

Reply:
Looks good, thanks 👍